lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2.cpp

```diff
@@ -346,13 +346,12 @@ bool MMU2::rx_str_P(const char *str) {
     uint8_t i = strlen(rx_buffer);
 
     while (mmuSerial.available()) {
-        rx_buffer[i++] = mmuSerial.read();
-        rx_buffer[i] = '\0';
-
-        if (i == sizeof(rx_buffer) - 1) {
+        if (i == sizeof(rx_buffer) - 2) {
             DEBUG_ECHOLNPGM("rx buffer overrun");
             break;
         }
+        rx_buffer[i++] = mmuSerial.read();
+        rx_buffer[i] = '\0';
```

Comment:
The terminating null set can as well be put outside of the loop.

Reply:
absolutely. good point.